src/java.base/share/classes/java/lang/reflect/code/op/CoreOp.java

```diff
@@ -1465,6 +1444,77 @@ public boolean hasReceiver() {
         public TypeElement resultType() {
             return resultType;
         }
+
+        static MethodRef createInvokeDescriptor(ExternalizedOp def) {
+            return def.extractAttributeValue(ATTRIBUTE_INVOKE_DESCRIPTOR,
+                    true, v -> switch (v) {
+                        case String s -> MethodRef.ofString(s);
+                        case MethodRef md -> md;
+                        case null, default -> throw new UnsupportedOperationException("Unsupported invoke descriptor value:" + v);
+                    });
+        }
+
+
+        /**
+         * The invoke instance or class (static) operation, that can model Java language instance or class method
+         * invocation expressions.
+         */
+        @OpFactory.OpDeclaration(InvokeInstanceClassOp.NAME)
+        public static final class InvokeInstanceClassOp extends InvokeOp {
```

Comment:
Unsure whether this is just a kind - e.g. whether to split or lump. So I'm ok with what you did for now.